### ISSUE TITLE
Option to change how expiry create and access at same time work

### ISSUE THREAD

**Issue by devnopt** [ehcache/ehcache3#2909]
When using create (TTL) and access (TTI) at the same time it appears to wait until both conditions occur before expiring. I think there should be an option to expire the entry once either condition is met.

Thanks.

**Comment by chrisdennis** [comment]
TTI is officially informally deprecated as a concept in Ehcache 3. You can read the rationale for that decision here: https://github.com/ehcache/ehcache3/issues/1097

That said we should explore the behavior of the feature you describe. The diagram below shows the lifespan of a cache entry that is created at 'CREATE' and subsequently accessed at 'ACCESS' under both the current behavior and my interpretation of your proposal for two different TTI values in combination with a TTL.
<img width="599" alt="ttl tti" src="https://user-images.githubusercontent.com/954331/119899598-4f41e580-bf11-11eb-97d6-b2e331efe58c.png">

There are four overall possible combinations of behavior:

### P1 & P2
This is your proposal. In this mode the act of reading from the cache can cause the read element to expire earlier than it would if you didn't read it. This implies that somehow reading something either a) makes it 'dirtier' sooner, or b) makes it a candidate for eviction - reading something now makes me somehow less likely to need it in the future?

### C1 & C2
This is just the existing behavior.

Now the crosses:
### C1 & P2
This is the same as just regular TTL.

### P1 & C2
This is even weirder. In this model the act of reading something totally replaces the TTL. The only reason I can see this making sense is if TTL and TTI are equal... but thats just using straight TTI. I still think TTI is strange for Ehcache, TTI makes little sense if you have a better method of capacity control.

In summary I can't see how this feature would be useful to a user. If I misunderstood the behavior you wanted, or there is some scenario I'm not seeing, please let me know. I'm going to close this for now though.

**Comment by devnopt** [comment]
The current implementation allows for indefinite stale data as long as data is continually pulled within the TTI, regardless of the TTL.

I think it is important to have the ability to evict an entry from the cache sooner than TTL, if it hasn't been read recently. This will reduce the potential eviction of more important data and cause the data not read recently to be refreshed sooner. 

Also, the TTI for the access/read operation should start from the moment the data is created (or be configurable to do so), not on the first read from the cache. When an application goes to get an entry and a miss occurs, it triggers a read to the data source followed by a write to the cache.

**Comment by chrisdennis** [comment]
The solution to these problems is to set your TTL according to how stale
you will tolerate your data being. Then turn off TTI and use capacity
constraints to control the size of your cache, not the TTI.
